### ISSUE TITLE
"Open in Terminal" menu option should open in PowerShell or Ubuntu depending on folder location

### ISSUE THREAD

**Issue by satvikpendem** [microsoft/terminal#14753]
# Description of the new feature/enhancement

Not sure if this is an enhancement versus an actual bug, but if have my default WSL distro be Ubuntu, and I click "Open in Terminal" in a Windows folder, I expect it to open up the *PowerShell* terminal, not the Ubuntu one. Vice versa if I have the default terminal as PowerShell and I click "Open in Terminal" in an Ubuntu folder, like in the below folders.

In other words, Terminal should know the context in which I'd want each terminal to open up.

![image](https://user-images.githubusercontent.com/42670561/215248711-7115c3e4-071a-432c-aa64-f942e98fdc10.png)

# Proposed technical implementation details (optional)

**Comment by zadjii-msft** [comment]
This is a neat idea! However, I think we'd probably end up getting stuck trying to pick the "right" profile for each scenarios. E.g. what about someone with both Windows PowerShell and PowerShell 7? Or multiple WSL distros?

I think it'll be easier for us to just let folks pick the profile they want to open instead, That's tracked in /dup #6111

**Comment by DHowett** [comment]
I actually think this is really clever and worth discussing more :smile:

If the user has our stock profiles or any of the autodetected ones, we can absolutely know which distribution should be used to launch in which WSL directory.

**Comment by zadjii-msft** [comment]
I'm confident that us guessing means we'll guess wrong.

* what about folks with the Canonical WSL profiles?
* What about little old me with my Command Prompt default profile?
* What about me with my non-built-in default profile?
* What if I have a WSL set as the default, and I _do_ want to open WSL navigated to %USERPROFILE%, so I can do ~sane~ bash scripting on my Windows files?

**Comment by DHowett** [comment]
Bullet point 4 is a bit beside the point - the shell extension changes the `startingDirectory` somewhat _by design._ :smile:

For the rest: guessing wrong is low cost and I would posit rare. I guess the question is...

Is it more delightful for us to launch `\\wsl\Debian\` paths into Debian under WSL automatically (before we have profile selection!) than it is annoying for us to not launch your default profile?

**Comment by satvikpendem** [comment]
I also suspect most users have the default PowerShell and one WSL profile, so guessing correctly for most users would probably be accurate.

**Comment by OldStarchy** [comment]
Even if the guess is the wrong version of powershell (assuming you have more than one version installed) it would still at least give you a terminal in the correct directory. Currently the "Open in Terminal" option is 100% useless to me because it opens in WSL so I still have to go through roughly the same steps to open another terminal with powershell and navigate to the correct directory.

Opening the wrong powershell can more easily fixed by just manually running the correct pwsh command.
